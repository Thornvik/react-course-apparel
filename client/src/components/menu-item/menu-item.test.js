import React from "react";
import { shallow } from "enzyme";
import { MenuItem } from "./menu-item.component";

describe("menuItem Component", () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const linkUrl = "/hats";
  const size = "large";
  const imageUrl = "testimage";

  beforeEach(() => {
    mockMatch = {
      url: "/shop",
    };

    mockHistory = {
      push: jest.fn(),
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      linkUrl,
      size,
      title: "hats",
      imageUrl,
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  it("should render menu-item component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push with correct string when menu-item is clicked", () => {
    wrapper.simulate("click");

    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
  });

  it("should Pass size to menu-item as class name", () => {
    expect(wrapper.prop("className")).toBe(`${size} menu-item`);
  });

  it("should pass the imageurl to Background image div", () => {
    expect(wrapper.find(".background-image").prop("style")).toHaveProperty(
      "backgroundImage",
      `url(${imageUrl})`
    );
  });
});
