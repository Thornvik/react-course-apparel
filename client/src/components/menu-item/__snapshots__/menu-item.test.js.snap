// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menuItem Component should render menu-item component 1`] = `ShallowWrapper {}`;
